module/VuFind/src/VuFind/Auth/AlmaDatabase.php

```diff
@@ -0,0 +1,161 @@
+<?php
+/**
+ * Alma Database authentication class
+ *
+ * PHP version 5
+ *
+ * Copyright (C) AK Bibliothek Wien für Sozialwissenschaften 2018.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Authentication
+ * @author   Michael Birkner <michael.birkner@akwien.at>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:plugins:authentication_handlers Wiki
+ */
+
+namespace VuFind\Auth;
+
+use VuFind\Exception\Auth as AuthException;
+use Zend\Crypt\Password\Bcrypt;
+
+class AlmaDatabase extends Database
+{
+    
+    /**
+     * ILS Authenticator
+     *
+     * @var \VuFind\Auth\ILSAuthenticator
+     */
+    protected $authenticator;
+    
+    /**
+     * Catalog connection
+     *
+     * @var \VuFind\ILS\Connection
+     */
+    protected $catalog = null;
+    
+    /**
+     * Alma driver
+     * 
+     * @var \VuFind\ILS\Driver\Alma
+     */
+    protected $almaDriver = null;
+    
+    
+    /**
+     * Alma config
+     * 
+     * @var array
+     */
+    protected $almaConfig = null;
+    
+    /**
+     * Constructor
+     * 
+     * @param \VuFind\ILS\Connection $connection
+     * @param \VuFind\Auth\ILSAuthenticator $authenticator
+     */
+    public function __construct(\VuFind\ILS\Connection $connection, \VuFind\Auth\ILSAuthenticator $authenticator)
+    {
+        $this->catalog = $connection;
+        $this->authenticator = $authenticator;
+        $this->almaDriver = $connection->getDriver();
+        $this->almaConfig = $connection->getDriverConfig();
+    }
+    
+    /**
+     * Create a new user account in Alma AND in the VuFind Database.
+     * 
+     * @param \Zend\Http\PhpEnvironment\Request $request Request object containing new account details.
+     * @return NULL|\VuFind\Db\Row\User New user row.
+     */
+    public function create($request)
+    {    	
+    	// When in privacy mode, don't create an Alma account and delegate
+    	// further code execution to the parent.
+    	if ($this->getConfig()->Authentication->privacy) {
+    		return parent::create($request);
+    	}
+    	
+        // User variable
+        $user = null;
+        
+        // Ensure that all expected parameters are populated to avoid notices
```

Comment:
It might be worth refactoring this code a little bit to reduce the duplication of code from the base Database driver. I could imagine a collectParamsFromRequest() method to encapsulate the current lines 97-105 and a validateParams() method to encapsulate lines 107-127. (Obviously you'd want to refactor these in the base Database class).